crates/ruff_python_parser/src/snapshots/ruff_python_parser__lexer__tests__repeated_string_prefix_error.snap

```diff
@@ -0,0 +1,34 @@
+---
+source: crates/ruff_python_parser/src/lexer.rs
+expression: "lex(r#\"uU\"hello\"\"#, Mode::Module, TextSize::default())"
+---
+## Tokens
```

Comment:
Let's also add a parser test. I'm curious to see how the parser recovers overall.

Reply:
> I don't think this trips up the parser, and we don't do anything downstream with source that has invalid syntax - i.e. no lint rules are run - so I don't think we have to worry.

I think we do run all token based rules. 

> But I could try to arrange for the range to exclude the prefix? Then the AST would behave as if there was nothing there - like it does when we skip trivia.

We would need to emit the prefix as an `Unknown` token because the lexer isn't allowed to skip over **any** content. I suspect that this will undo much of the improvements you made (as the `Unknown` token will confuse the parser)

Reply:
That's certainly worth exploring. One challenge is that the data needs to fit into `TokenFlags`, which is a `u8`. The other, that you still need to pick a `TokenKind`. 

The last challenge is that it requires designing how the different `StringFlags` represent this new state (most of those currently assert that the flags are valid). 

I'm not sure I can help you much with this right now as it seems fairly low priority (while a nice improvement). 

For now, I suggest making a local improvement (which might be to go with `Unknown`)